src/components/sequencing/SequenceEditor.svelte

```diff
@@ -274,6 +285,77 @@
     }
   }
 
+  const blockMark = Decoration.mark({ class: 'cm-block-match' });
+
+  const blockTheme = EditorView.baseTheme({
+    '.cm-block-match': {
+      outline: '1px dashed',
+    },
+  });
+
+  function highlightBlock(viewUpdate: ViewUpdate) {
+    const tree = syntaxTree(viewUpdate.state);
+    // Command Node includes trailing newline and white space, move to next command
+    const selectionLine = viewUpdate.state.doc.lineAt(viewUpdate.state.selection.asSingle().main.from);
+    const leadingWhiteSpaceLength = selectionLine.text.length - selectionLine.text.trimStart().length;
+    const updatedSelectionNode = tree.resolveInner(selectionLine.from + leadingWhiteSpaceLength, 1);
+    const stemNode = getNearestAncestorNodeOfType(updatedSelectionNode, [TOKEN_COMMAND])?.getChild('Stem');
+
+    if (!stemNode || !isBlockCommand(viewUpdate.state.sliceDoc(stemNode.from, stemNode.to))) {
+      return [];
+    }
+
+    const blocks = computeBlocks(viewUpdate.state);
+    if (!blocks) {
+      return [];
+    }
+
+    const pairs = Object.values(blocks);
+    const matchedNodes: SyntaxNode[] = [stemNode];
+
+    // when cursor on end -- select else and if
+    let current: SyntaxNode | undefined = stemNode;
+    while (current) {
+      current = pairs.find(block => block.end?.from === current!.from)?.start;
+      if (current) {
+        matchedNodes.push(current);
+      }
+    }
+
+    // when cursor on if -- select else and end
+    current = stemNode;
+    while (current) {
+      current = pairs.find(block => block.start?.from === current!.from)?.end;
+      if (current) {
+        matchedNodes.push(current);
+      }
+    }
+
+    return matchedNodes;
+  }
+
+  const blockHighlighter = ViewPlugin.fromClass(
+    class {
+      decorations: DecorationSet;
+      constructor(_view: EditorView) {
```

Comment:
can we also remove the `_view` argument if it's not being used or do you think it's useful to have it here for future usage?